### DISCUSSION TITLE
Is it possible to dynamically set the size of a popup

### DISCUSSION THREAD

**Discussion by davefxy** [CommunityToolkit/Maui#2022] [General]
I am displaying a varying length list and I want the length of the popup to be dependent on the number of items with a minimum length and a maximum length of the popup?

**Comment by stephenquan** [comment]
You can create a transparent full-screen popup, and, within it, you can control the children's size. e.g.

 - Popup.HorizontalOptions="Fill"
 - Popup.VerticalOptions="Fill"
 - Color="Transparent"

```xaml
<?xml version="1.0" encoding="utf-8" ?>
<toolkit:Popup
    x:Class="MauiVariablePopup.ResizePopup"
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
    HorizontalOptions="Fill"
    VerticalOptions="Fill"
    Color="Transparent">
    <Grid>
        <Grid
            BackgroundColor="SkyBlue"
            HeightRequest="{Binding Value, Source={Reference slider}}"
            WidthRequest="400">
            <Slider
                x:Name="slider"
                Maximum="400"
                Minimum="100"
                VerticalOptions="Center" />
        </Grid>
    </Grid>
</toolkit:Popup>
```

**Comment by sjordanGSS** [comment]
```
<?xml version="1.0" encoding="utf-8" ?>
<toolkit:Popup
    x:Class="MauiVariablePopup.ResizePopup"
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit">
  <ScrollView VerticalOptions="Start">
    <VerticalStackLayout BindableLayout.ItemsSource="{Binding [item list here]}">
    </VerticalStackLayout>
  </ScrollView>
</toolkit:Popup>
```

Something like this worked for me. The ```Popup``` should size vertically to the ```ScrollView``` which will size to the ```VerticalStackLayout```, and grow/shrink as items are added/removed. ```Popup``` doesn't have ```MinimumHeightRequest``` and ```MaximumHeightRequest``` properties and won't respect them on it's ```Content```, but if you wrap the ```ScrollView``` in a ```ContentView```, then set those properties on the ```ScrollView``` they should work.

**Reply by stephenquan** [reply]
When I tried your code against CommunityToolkit.Maui 9.0.2, the Popup was full-screen. This can be evidenced by setting `Popup.Color` to a visible color such as `"SkyBlue"` and the entire screen went that color. Whilst the Popup was up I was able to append items to my ObservableCollection and the BindableLayout dynamic changed to it so one would see the VerticalStackLayout to be growing or shrinking within the ScrollView. The only workaround I could see was to set the `Popup.Color` to `"Transparent"`.

**Reply by sjordanGSS** [reply]
Interesting, this is untested code based off of code that I have in a closed project where the ```ScrollView``` is contained within a ```Grid``` with ```<RowDefinition Height="*" />``` applied to it, perhaps that is affecting it.

**Reply by sjordanGSS** [reply]
I've just tested with the following code: 

```
<?xml version="1.0" encoding="utf-8" ?>
<toolkit:Popup xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
             x:Class="PopupContentMaximumHeightIgnored.Dialog"
               Color="Pink">
    <ScrollView VerticalOptions="Start"
                        WidthRequest="400">
          <VerticalStackLayout>
              <Label Text="Item 1" />
              <Label Text="Item 2" />
              <Label Text="Item 3" />
          </VerticalStackLayout>
      </ScrollView>
</toolkit:Popup>
```

And this seems to work on Android 7.1. Only the area around the text appears pink. The fixed items in the ```VerticalStackLayout``` shouldn't make a difference to sizing, as ```BindableLayout``` is just inserting views into the target view with the ```BindingContext``` changed.

**Comment by CliffAgius** [comment]
Closed as answered, if closed in error please let us know to re-open.
